### DISCUSSION TITLE
An error was encountered while using .to_homogeneous()

### DISCUSSION THREAD

**Discussion by RENEK-bool** [pyg-team/pytorch_geometric#9308] [Q&A] [answered]
I am currently planning to use HEATConv. I have read some discussions in the past (https://github.com/pyg-team/pytorch_geometric/discussions/6869) and found that I need to perform `hetero_data.to_homogeneous()` first to get the correct input format, but when I use `.to_homogeneous()` an error occurs.
```
wire_train_sample = train_dataset[0].to_homogeneous()
RuntimeError: Sizes of tensors must match except in dimension 0. Expected size 1 but got size 5 for tensor number 1 in the list.
```
This is my HeteroData format
```
HeteroData(
  pin={ x=[66704, 1] },
  (pin, cell_arc, pin)={
    edge_index=[2, 43152],
    edge_attr=[43152, 1],
  },
  (pin, net_arc, pin)={
    edge_index=[2, 42662],
    edge_attr=[42662, 5],
    y=[42662],
  }
)
```
Is it because I put the label y on the edge (pin, net_arc, pin) that caused the error?
In addition, I saw that HeteroData.to_homogenous() will change the edge order.(https://github.com/pyg-team/pytorch_geometric/issues/8707) Will this cause my edge labels to not correspond? What I'm currently trying to achieve is edge regression.

**Comment by rusty1s** [comment, marked as answer]
This is probably because `edge_attr` is of different size in the two edge types. What you can do is to "pad" each `edge_attr` to size `[num_edges, 6]`.
